### PR TITLE
implements update API and client code

### PR DESCRIPTION
**Pull request by flyjwayur**

### CODE REVIEW

File: scripts/main.js [hunk 1]
```diff
@@ -100,6 +110,10 @@ function renderAppDetail (id) {
 
   .then(function (app) {
     $('#name-app').html(app.name)
+    // $('#name-app').data('id', ""+app.id)
+    $('#name-app').attr('data-id', ""+app.id)
+    // $('#name-app').prop('data-id', ""+app.id)
+    // console.log(app.id)
```
Comment (gr2m) [inline scripts/main.js:116]: once things work, I’d remove the comments like this, to keep the code clean

File: scripts/indexApi.js [hunk 2]
```diff
@@ -25,3 +25,26 @@ applist.find = function (id) {
...
 }
+applist.add = function (app) {
+  app.id = Math.random().toString(36).substr(2, 7)
+  return applist.findAll()
+    .then(function (apps) {
+      apps.push(app)
+      localStorage.setItem('apps', JSON.stringify(apps))
+      return app
+    })
+}
```
Comment (gr2m) [inline scripts/indexApi.js:36]: did you add `applist.add` again by accident? It’s now defined twice

**Comment by titay2** [comment]
it must be by accident the code works fine after the second one is removed

File: scripts/main.js [hunk 3]
```diff
@@ -100,6 +110,10 @@ function renderAppDetail (id) {
 
   .then(function (app) {
     $('#name-app').html(app.name)
+    // $('#name-app').data('id', ""+app.id)
```
Comment (flyjwayur) [inline scripts/main.js:113]: @gr2m @titay2 Thank you for the comments and feedback :D I'll clean up as soon as possible after solving one question. I was able to get 'app.id' by attr('data-id', ""+app.id), but with '.data('id', ""+app.id)'  and .prop('data-id', ""+app.id)', I was not able to get data-id. I thought three codes would work same, but it didn't. I wonder what are the mistakes, when I used '.data()' and '.prop()'. :)

File: scripts/indexApi.js [hunk 4]
```diff
@@ -25,3 +25,26 @@ applist.find = function (id) {
       return apps.find(findApp)
     })
 }
+applist.add = function (app) {
```
Comment (flyjwayur) [inline scripts/indexApi.js:28]: @gr2m @titay2 By accident, I added the applist.add twice :D Thank you for the feedback and fixing that part :) Yeahhyyy, happy to have the best teamwork! :D

**Pull request merged**